reconcile/slack_usergroups.py

```diff
@@ -298,6 +305,34 @@ def get_desired_state(slack_map, pagerduty_map):
     return desired_state
 
 
+def _create_usergroups(
+    current_ug_state: dict,
+    desired_ug_state: dict,
+    slack_client: SlackApi,
+    dry_run: bool = True,
+):
+    """
+    Create Slack usergroups.
+    """
+
+    workspace = desired_ug_state["workspace"]
+    usergroup = desired_ug_state["usergroup"]
+
+    if current_ug_state:
+        logging.debug(
+            f"[{workspace}] Usergroup exists and will not be created {usergroup}"
+        )
+        return
+
+    logging.info(["create_usergroup", workspace, usergroup])
+    if not dry_run:
+        try:
+            usergroup_id = slack_client.create_usergroup(usergroup)
+            desired_ug_state["usergroup_id"] = usergroup_id
```

Comment:
once the usergroup is created, update the usergroup id in the desired state for following actions (adding users, etc)